components/event-sources/reconciler/httpsource/reconcile.go

```diff
@@ -144,6 +158,33 @@ func (r *Reconciler) reconcileSink(src *sourcesv1alpha1.HTTPSource) (*messagingv
 	return currentCh, nil
 }
 
+// reconcilePolicy reconciles the state of the Policy.
+func (r *Reconciler) reconcilePolicy(src *sourcesv1alpha1.HTTPSource, ksvc *servingv1alpha1.Service) (*authenticationv1alpha1.Policy, error) {
+	if ksvc == nil {
+		r.event(src, failedCreateReason, "Skipping creation of Policy as there is no ksvc yet")
+		return nil, nil
+	}
+
+	if &ksvc.Status != nil && len(ksvc.Status.ConfigurationStatusFields.LatestCreatedRevisionName) == 0 {
```

Comment:
By the way, `ksvc.Status` is of type `ServiceStatus` whose address shouldn't be nil in order to further check ksvc.Status.foo Why do you think idiomatic string comparison makes sense here?

Reply:
`ServiceStatus` is a nested _struct_, not _pointer to struct_. Its zero value is `ServiceStatus{}`, it will _never_ be nil.

Regarding the string, the general idea is simply "make the code clear". I tend to think `len()` is more appropriate for collections (slices, maps, ...), whereas `== ""` clearly tells "this is a string, and it's empty". Both a technically the same.